scripts/api_dump_generator.py

```diff
@@ -439,6 +439,37 @@
 }}
 @end function
 
+VK_LAYER_EXPORT VKAPI_ATTR VkResult VKAPI_CALL vkGetPhysicalDeviceToolPropertiesEXT(VkPhysicalDevice physicalDevice, uint32_t *pToolCount, VkPhysicalDeviceToolPropertiesEXT *pToolProperties)
+{{
+    dump_head_vkGetPhysicalDeviceToolPropertiesEXT(ApiDumpInstance::current(), physicalDevice, pToolCount, pToolProperties);
+    static const VkPhysicalDeviceToolPropertiesEXT api_dump_layer_tool_props = {{
+        VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_TOOL_PROPERTIES_EXT,
+        nullptr,
+        "API Dump Layer",
+        "2",
```

Comment:
This is version "2", the others are version "1"?

Reply:
I was looking at the "Layer Version" perimeter output by vulkaninfo because I thought it matched the "implementation_version" in the layer JSON manifest files.

Looks like they aren't quite the same.

Do you or anyone else know the difference between the "implementation_version" and the "Layer Version"?

Reply:
There are several versions at play, I'm not clear on the semantic differences:
1. JSON's implementation_version
2. VkPhysicalDeviceToolPropertiesEXT::version (char[])
3. VkLayerProperties::implementationVersion (uint32_t)

Reply:
Do you think that this is important enough to hold up the PR until this is settled, or should I merge and we use another PR to update this later?

Reply:
It would be best to have the versioning resolved before the next SDK.  For the Vk versions, if the spec doesn't have guidance, then ping the WG list?  Who is best resource for the JSON version, a loader person?

The idea is to have it correct before anyone begins to depend on it.

Reply:
From the spec:

> version is a null-terminated UTF-8 string containing the version of the tool.

So the question is, what most accurately describes the version of a layer, in this case, Device Simulation.

Reply:
The DevSim version is defined [here](https://github.com/LunarG/VulkanTools/blob/master/layersvt/device_simulation.cpp#L69-L72).

IMO I'd propose all three of the versions above be set to the same value.

We need a mechanism to derive the versions from a single point of truth.  Really seems like the ToolProperties and JSON versions should somehow be derived from LayerProperties.

In lieu of mechanism, for the time being perhaps blatant commenting to remind editors to ensure version synchronization in the future?

Reply:
Updated `VK_LAYER_LUNARG_device_simulation` to derive the version from the major-minor-patch variables you highlighted.